src/yang/ietf-bgp-policy.yang

```diff
@@ -248,6 +248,32 @@ module ietf-bgp-policy {
           }
         }
       }
+      container next-hop-sets {
+        description
+          "Definition of a list of IPv4 or IPv6 next-hops which can
+           be matched in a routing policy.";
+
+        list next-hop-set {
+          key "name";
+          description
+            "List of defined next-hop sets for use in policies.";
+
+          leaf name {
+            type string {
+              length "1..79";
+            }
+            mandatory true;
+            description
+              "Name of the next-hop set.";
+          }
+          leaf-list address {
+            type bgp-next-hop-type;
```

Comment:
I think this type will _probably_ be fine.  It's scoped as the "no-zone" one.
The thing I'm concerned about is implementations that only want matching address family elements for a given session type.  For example, for ipv4 unicast we should only have nexthops that are ipv4.

Juniper's implementation, for example, had a period where matched nexthops were a requirement.  Later, when support for the various v4 over v6 or vice versa features were introduced, a mix was possible.

This is a challenging IETF implementation-neutral question.  It may simply be easiest to not constrain for address family matching and let implementations enforce it at commit time.

I'd appreciate Keyur's thinking here too.

Reply:
IMHO, we should perform a check of same session supporting same nexthop-type at the back end as part of validation. Most implementations today support v4 over v6 and vice versa.

Reply:
One more comment.  since its match next-hop I would suggest keeping next-hop type as inept:ip-address as opposed to no-zone-ip-address?

Reply:
> One more comment. since its match next-hop I would suggest keeping next-hop type as inept:ip-address as opposed to no-zone-ip-address?

Zone is not present in the PDUs; it's an inferred property of the session.  That's why it should be excluded.